modules/document_repository/php/editfile.class.inc

```diff
@@ -0,0 +1,183 @@
+ */
+namespace LORIS\document_repository;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * It includes two functions.
+ * Post method handles updating a file.
+ * Delete method handles deleting a file.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Document_Repository
+ * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class EditFile extends \NDB_Page
+{
+    public $skipTemplate = true;
+
+    /**
+     * Same as the permissions for the main menu filter page.
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool true iff the user has access to this page.
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return (
+            $user->hasPermission('document_repository_delete')
+            || $user->hasPermission('document_repository_view')
+
+        );
+    }
+
+    /**
+     * Handle a post form
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        switch ($request->getMethod()) {
+        case "POST":
+            $this->editDocFile($request);
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withStatus(200)
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode("update")
+                    )
+                );
+          break;
+        case "DELETE":
+            $path = $request->getQueryParams();
+            $id   = basename($path['lorispath']);
+            $this->deleteFile($id);
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withStatus(200)
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode("delete")
+                    )
+                );
+          break;
+        default:
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withStatus(405)
+                ->withHeader("Allow", "POST and DELETE");
+        }
+    }
+    /**
+ * Handles the document editing process
+ *
+ * @param ServerRequestInterface $request The incoming PSR7 request
+ *
+ * @throws \DatabaseException
+ *
+ * @return void
+ */
+    function editDocFile($request): void
+    {
+        $db           = \Database::singleton();
+        $user         = \User::singleton();
+        $req          = $request->getParsedBody();
+        $updateValues = [
+                         'File_category' => isset($req['category']) ?
```

Comment:
these can all be the null coalesce operator: ie `File_category' => $req['category'] ?? null`